plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -108,6 +108,83 @@ func addClusterRoleBindingLabel(rolebindings []rbacv1.ClusterRoleBinding) {
 	return
 }
 
+func viewRules() []rbacv1.PolicyRule {
+	rules := []rbacv1.PolicyRule{
+		rbacv1helpers.NewRule(Read...).Groups(legacyGroup).Resources("pods", "replicationcontrollers", "replicationcontrollers/scale", "serviceaccounts",
+			"services", "services/status", "endpoints", "persistentvolumeclaims", "persistentvolumeclaims/status", "configmaps").RuleOrDie(),
+		rbacv1helpers.NewRule(Read...).Groups(legacyGroup).Resources("limitranges", "resourcequotas", "bindings", "events",
+			"pods/status", "resourcequotas/status", "namespaces/status", "replicationcontrollers/status", "pods/log").RuleOrDie(),
+		// read access to namespaces at the namespace scope means you can read *this* namespace.  This can be used as an
+		// indicator of which namespaces you have access to.
+		rbacv1helpers.NewRule(Read...).Groups(legacyGroup).Resources("namespaces").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(discoveryGroup).Resources("endpointslices").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(appsGroup).Resources(
+			"controllerrevisions",
+			"statefulsets", "statefulsets/status", "statefulsets/scale",
+			"daemonsets", "daemonsets/status",
+			"deployments", "deployments/status", "deployments/scale",
+			"replicasets", "replicasets/status", "replicasets/scale").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(autoscalingGroup).Resources("horizontalpodautoscalers", "horizontalpodautoscalers/status").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(batchGroup).Resources("jobs", "cronjobs", "cronjobs/status", "jobs/status").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(extensionsGroup).Resources("daemonsets", "daemonsets/status", "deployments", "deployments/scale", "deployments/status",
+			"ingresses", "ingresses/status", "replicasets", "replicasets/scale", "replicasets/status", "replicationcontrollers/scale",
+			"networkpolicies").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(policyGroup).Resources("poddisruptionbudgets", "poddisruptionbudgets/status").RuleOrDie(),
+
+		rbacv1helpers.NewRule(Read...).Groups(networkingGroup).Resources("networkpolicies", "ingresses", "ingresses/status").RuleOrDie(),
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
+		rules = append(rules, rbacv1helpers.NewRule(Read...).Groups(resourceGroup).Resources("resourceclaims", "resourceclaims/status", "resourceclaimtemplates").RuleOrDie())
```

Comment:
I think we should also grant read permission for "deviceclasses" and "resourceslices".

Or is that handled somehow elsewhere? They are cluster-scoped.

They are listed below in plugin/pkg/auth/authorizer/rbac/bootstrappolicy/testdata/cluster-roles-featuregates.yaml

Reply:
Cluster scoped resources are not in scope for the `admin`, `edit`, `view` cluster roles which are only expected to be bound via a role binding in specific namespaces.  We don't add any RBAC for such resources outside of what is needed for nodes / scheduler / controllers / etc.

Reply:
So how does a user in, say, GKE get read permissions for "deviceclasses" and "resourceslices"?

Do they have it automatically as soon as they exist, or is that something that GKE has to set up?

Reply:
GKE/AKS/EKS/etc gives the customer (who created the cluster) "cluster admin" access which allows everything (authorization is pluggable and there are numerous ways to accomplish this).  The customer is free to set up new RBAC cluster roles and cluster role bindings with whatever permissions they want to assign to their users.